From 0e07936f15b7f4fda57d31f7b30ddba8b3f72fa7 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Tue, 21 May 2013 02:05:51 +0200
Subject: [PATCH 283/509] dma: xilinx: vdma: Remove disable_intr support

https://github.com/analogdevicesinc/linux.git xcomm_zynq
commit 83c96d70d4a80cba4f1ebaf7738eae2b2f599f86

The client must not control whether interrupts should be enabled. Remove
that.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Acked-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 drivers/dma/xilinx/vdmatest.c       |    1 -
 drivers/dma/xilinx/xilinx_axivdma.c |   17 +++--------------
 include/linux/amba/xilinx_dma.h     |    1 -
 3 files changed, 3 insertions(+), 16 deletions(-)

diff --git a/drivers/dma/xilinx/vdmatest.c b/drivers/dma/xilinx/vdmatest.c
index f4debd1..9e18361 100644
--- a/drivers/dma/xilinx/vdmatest.c
+++ b/drivers/dma/xilinx/vdmatest.c
@@ -306,7 +306,6 @@ static int vdmatest_slave_func(void *data)
 		config.gen_lock = 0;
 		config.master = 0;
 		config.park_frm = 0;
-		config.disable_intr = 0;
 		tx_dev->device_control(tx_chan, DMA_SLAVE_CONFIG,
 					(unsigned long)&config);
 
diff --git a/drivers/dma/xilinx/xilinx_axivdma.c b/drivers/dma/xilinx/xilinx_axivdma.c
index 2dcc212..24c4ed2 100644
--- a/drivers/dma/xilinx/xilinx_axivdma.c
+++ b/drivers/dma/xilinx/xilinx_axivdma.c
@@ -81,7 +81,6 @@
 #define XILINX_VDMA_XR_COALESCE_MASK	0x00FF0000
 						/* Coalesce counter */
 
-#define XILINX_VDMA_IRQ_SHIFT		12
 #define XILINX_VDMA_DELAY_SHIFT		24
 #define XILINX_VDMA_COALESCE_SHIFT	16
 
@@ -515,17 +514,9 @@ static void xilinx_vdma_start_transfer(struct xilinx_vdma_chan *chan)
 	 * Enable interrupts
 	 * park/genlock testing does not use interrupts
 	 */
-	if (!chan->config.disable_intr) {
-		VDMA_OUT(&chan->regs->cr,
-			VDMA_IN(&chan->regs->cr) |
-				XILINX_VDMA_XR_IRQ_ALL_MASK);
-	} else {
-		VDMA_OUT(&chan->regs->cr,
-			(VDMA_IN(&chan->regs->cr) |
-				XILINX_VDMA_XR_IRQ_ALL_MASK) &
-			~((chan->config.disable_intr <<
-				XILINX_VDMA_IRQ_SHIFT)));
-	}
+	VDMA_OUT(&chan->regs->cr,
+		VDMA_IN(&chan->regs->cr) |
+			XILINX_VDMA_XR_IRQ_ALL_MASK);
 
 	/* Start the transfer */
 	if (chan->has_sg)
@@ -1008,8 +999,6 @@ static int xilinx_vdma_device_control(struct dma_chan *dchan,
 			chan->config.delay = cfg->delay;
 		}
 
-		chan->config.disable_intr = cfg->disable_intr;
-
 		/* FSync Source selection */
 		reg &= ~XILINX_VDMA_CR_FSYNC_SRC_MASK;
 		reg |= cfg->ext_fsync << XILINX_VDMA_EXTFSYNC_SHIFT;
diff --git a/include/linux/amba/xilinx_dma.h b/include/linux/amba/xilinx_dma.h
index 72c1544..7b99b55 100644
--- a/include/linux/amba/xilinx_dma.h
+++ b/include/linux/amba/xilinx_dma.h
@@ -63,7 +63,6 @@ struct xilinx_vdma_config {
 	int park_frm;			/* Frame to park on */
 	int coalesc;			/* Interrupt coalescing threshold */
 	int delay;			/* Delay counter */
-	int disable_intr;		/* Whether use interrupts */
 	int reset;			/* Reset Channel */
 	int ext_fsync;			/* External Frame Sync */
 };
-- 
1.7.5.4

